pkg/util/proto/google_proto.go

```diff
@@ -89,10 +104,30 @@ func (o mergeOptions) mergeMessage(dst, src protoreflect.Message) {
 		panic(fmt.Sprintf("cannot merge into invalid %v message", dst.Descriptor().FullName()))
 	}
 
+	if o.replaceList {
+		fields := src.Descriptor().Fields()
+		// separate loop to support empty lists in 'src'
+		// because 'src.Range' doesn't return empty fields
```

Comment:
If Range is the problem should you just be implementing your own with src.Descriptor().Fields() instead of hacking around it?

Reply:
Implementing own Range means: 
* reimplementing `mergeMap`, `mergeMessage` and `dst.Set` because they don't expect `v` to be `nil`
* covering with tests not only `list` functionality but also other field types

Reply:
but there is bigger problem, protobuf doesn't make a difference between:
```
a: hello
b: []
```
and
```
a: hello
```
in both cases `b` is an empty list from the proto perspective

Reply:
so probably empty list in `src` should not override list in `dst`

Reply:
Ha right... What were the cases where we wanted this?